### PR TITLE
MSDKUI-549 Test reorder waypoints

### PR DESCRIPTION
**Pull request by Docik**
Reorder waypoints test implementation
Cleanup

Signed-off-by: Aleksandr Dotsenko <1225028+Docik@users.noreply.github.com>

### CODE REVIEW

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt [hunk 1]
```diff
@@ -94,6 +94,26 @@ open class CoreActions {
         ))
     }
 
+    /**
+     * An action that performs a swipe from center to [n] view heights
+     * above or below the view center. n > 0 - swipe down, n < 0 - swipe up
+     */
+    fun swipeHeightsUpDownCenter(n: Int): ViewAction {
```
[the remark this answers is not in the document]
Comment (Docik) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreActions.kt]: Yes, why not. Fixed.

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/views/routeplanner/useractions/RoutePlannerActions.kt [hunk 2]
```diff
@@ -156,6 +158,15 @@ object RoutePlannerActions {
         return this
     }
 
+    /**
+     * Drag one waypoint to another in waypoints list
+     */
+    fun dragWaypoint(waypoint: WaypointItem, to: WaypointItem) {
+        var n = to.value - waypoint.value
```
Comment (imvjk) [inline MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/views/routeplanner/useractions/RoutePlannerActions.kt]: can u please explain this a bit? value is index of waypoint? and may b also consider to rename "waypoint" to "from"?

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/routing/flow/RoutePlannerFlowTests.kt [hunk 3]
```diff
@@ -471,6 +484,65 @@ class RoutePlannerFlowTests : TestBase<SplashActivity>(SplashActivity::class.jav
...
+                if (wp1 != wp2) {
+                    // Drag waypoint
+                    RoutePlannerActions.dragWaypoint(wp1, wp2)
+                    // Reorder list in the same manner, as waypoints were reordered
+                    draggedItemIndex = wp1.value
+                    toIndex = wp2.value
+                    if (abs(draggedItemIndex - toIndex) == 2) {
+                        Collections.swap(waypointsList, draggedItemIndex, toIndex)
+                        Collections.swap(waypointsList, draggedItemIndex, 3 - draggedItemIndex - toIndex)
+                    } else {
```
Comment (imvjk) [inline MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/routing/flow/RoutePlannerFlowTests.kt]: what are you trying to do here?

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/views/routeplanner/useractions/RoutePlannerActions.kt (hunk 2 above)
Comment (Docik) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/views/routeplanner/useractions/RoutePlannerActions.kt]: This is a function to drag one waypoint item of waypoints list to another in the same list. How much is needed to drag is defined by n heights of the list item, where mod(n) = mod(index1 - index2) +1, and sign defines the direction (up or down).
As for naming, "from" will work for me as well. Fixed.

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/routing/flow/RoutePlannerFlowTests.kt (hunk 3 above)
Comment (Docik) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/routing/flow/RoutePlannerFlowTests.kt]: Keeping the List<String> waypointsList in the same order, as UI waypoints list after reordering items on UI

Comment (Docik) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/tests/routing/flow/RoutePlannerFlowTests.kt]: And the top most double cycle iterates thru all possible reorder combinations

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/views/routeplanner/useractions/RoutePlannerActions.kt (hunk 2 above)
Comment (Docik) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/views/routeplanner/useractions/RoutePlannerActions.kt]: And yes, value == index. I'll create a ticket to rename it.

**Comment by Docik** [comment]
WaypointItem.value renamed to WaypointItem.index, test logic comments added

**Pull request merged**